promql/engine_test.go

```diff
@@ -1745,6 +1749,262 @@ load 1ms
 	}
 }
 
+type simpleRangeQueryOpts struct {
+	start, end, interval                             int64
+	requirePrepError, requireQueryError, requireWarn bool
+}
+
+func simpleRangeQuery(t *testing.T, engine *promql.Engine, storage *teststorage.TestStorage, query string, opts *simpleRangeQueryOpts) promql.Matrix {
+	var err error
+	var qry promql.Query
+	qry, err = engine.NewRangeQuery(context.Background(), storage, nil, query, time.Unix(opts.start, 0), time.Unix(opts.end, 0), time.Duration(opts.interval)*time.Second)
+	if opts.requirePrepError {
+		require.Error(t, err)
+		return promql.Matrix(nil)
+	}
+	require.NoError(t, err)
+
+	res := qry.Exec(context.Background())
+	if opts.requireQueryError {
+		require.Error(t, res.Err)
+		return promql.Matrix(nil)
+	}
+	if opts.requireWarn {
+		require.NotEmpty(t, res.Warnings)
+	} else {
+		require.Empty(t, res.Warnings)
+	}
+	require.NoError(t, res.Err)
+	return res.Value.(promql.Matrix)
+}
+
+func requireCommonSeries(t *testing.T, expected, actual promql.Matrix, expectedLen int) {
+	commonSeries := 0
+	for _, actSeries := range actual {
+		for _, expSeries := range expected {
+			if testutil.CheckEqual(t, actSeries, expSeries) {
+				commonSeries++
+				if commonSeries == expectedLen {
+					return
+				}
+			}
+		}
+	}
+	require.Fail(t, fmt.Sprintf("Expected and Actual matrices don't have %d equal series:\n"+
+		"Expected:\n%s\n"+
+		"Actual:\n%s\n",
+		expectedLen, expected, actual))
+}
+
+func TestLimitK(t *testing.T) {
```

Comment:
Could this (and TestLimitRatio below) be done via the PromQL testing framework? See all the files in https://github.com/prometheus/prometheus/tree/main/promql/promqltest/testdata . Could be a new file `limit.test` there.